app/assets/javascripts/bookmarklet.js

```diff
@@ -0,0 +1,45 @@
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
+
+var bookmarklet = function(url, width, height, opts) {
+  // calculate popup dimensions & placement
+  var dim = function() {
+    var  w = window,
+        wT = w.screenTop ? w.screenTop : w.screenY,
+        wL = w.screenLeft ? w.screenLeft : w.screenX,
+         t = wT + (w.innerHeight / 2) - (height / 2),
+         l = wL + (w.innerWidth / 2) - (width / 2);
+    return "width=" + width + ",height=" + height + ",top=" + t + ",left=" + l;
+  };
+
+  // prepare url parameters
+  var params = function() {
+    var w = window,
+        d = document,
+        h = w.location.href,
+        t = d.title,
+      sel = w.getSelection ? w.getSelection() :
+            d.getSelection ? d.getSelection() :
+            d.selection.createRange().text,
+        n = sel.toString();
+    return "url=" + encodeURIComponent(h) +
+           "&title=" + encodeURIComponent(t) +
+           "&notes=" + encodeURIComponent(n);
+  };
+
+  // popup (or redirect) action
+  var act = function() {
+    var popup_opts = opts || "location=yes,links=no,scrollbars=yes,toolbar=no",
+        jump_url   = url + "?jump=yes";
+
+    window.open(url + "?" + params(), "diaspora_bookmarklet", popup_opts + dim()) ||
+    (window.location.href = jump_url + "&" + params())
```

Comment:
Identifier 'jump_url' is not in camel case.<br>Expected an assignment or function call and instead saw an expression.<br>Missing semicolon.